constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/helper/widget/Carousel.java

```diff
@@ -293,13 +314,43 @@ private void updateItems() {
             // mIndex should map to i == startIndex
             View view = mList.get(i);
             int index = mIndex + i - startIndex;
-            if (index < 0) {
-                updateViewVisibility(view, emptyViewBehavior);
-            } else if (index >= mAdapter.count()) {
-                updateViewVisibility(view, emptyViewBehavior);
+            if (infiniteCarousel) {
```

Comment:
oh what I meant was more something like:
if (infiniteCarousel) {
  index = index % mAdapter.count();
}

but that's fine.

Reply:
🚀